src/services/webhdfs/backend.rs

```diff
@@ -423,21 +423,34 @@ impl WebhdfsBackend {
 
     /// get redirect destination from 307 TEMPORARY_REDIRECT http response
     async fn follow_redirect(&self, resp: Response<IncomingAsyncBody>) -> Result<String> {
-        let bs = resp.into_body().bytes().await.map_err(|e| {
-            Error::new(ErrorKind::Unexpected, "redirection receive fail")
+        if resp.status() != StatusCode::TEMPORARY_REDIRECT {
+            let err = Error::new(ErrorKind::Unexpected, "redirection fail: not 307")
+                .with_context("service", Scheme::Webhdfs)
+                .set_permanent();
+            return Err(err);
+        }
+        let loc = match resp.headers().get(HeaderName::from_static("location")) {
+            Some(loc) => loc.to_str().map_err(|e| {
+                Error::new(
+                    ErrorKind::Unexpected,
+                    "redirection fail: not valid redirection",
+                )
                 .with_context("service", Scheme::Webhdfs)
                 .set_source(e)
-        })?;
-        let loc = serde_json::from_reader::<_, Redirection>(bs.reader())
-            .map_err(|e| {
-                Error::new(ErrorKind::Unexpected, "redirection fail")
-                    .with_context("service", Scheme::Webhdfs)
-                    .set_permanent()
-                    .set_source(e)
-            })?
-            .location;
-
-        Ok(loc)
+                .set_permanent()
+            })?,
+            None => {
```

Comment:
Is it a good name that we have a `parse_location` like other headers?

Reply:
We can move this to `http_utils` and name it to `parse_location`. 🤔